src/hooks/useGridDimensions.ts

```diff
@@ -17,26 +17,20 @@ export function useGridDimensions(): [
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
     if (ResizeObserver == null) return;
 
-    function saveDimensions() {
-      // Get dimensions without scrollbars.
-      // The dimensions given by the callback entries in Firefox do not substract the scrollbar sizes.
-      // https://bugzilla.mozilla.org/show_bug.cgi?id=1733042
-      const { clientWidth, clientHeight } = gridRef.current!;
+    const resizeObserver = new ResizeObserver((entries) => {
+      const size = entries[0].contentBoxSize[0];
       // TODO: remove once fixed upstream
       // we reduce width by 1px here to avoid layout issues in Chrome
       // https://bugs.chromium.org/p/chromium/issues/detail?id=1206298
-      setGridWidth(clientWidth - (devicePixelRatio % 1 === 0 ? 0 : 1));
-      setGridHeight(clientHeight);
-    }
-
-    saveDimensions();
-    const resizeObserver = new ResizeObserver(saveDimensions);
+      setInlineSize(size.inlineSize - (devicePixelRatio % 1 === 0 ? 0 : 1));
```

Comment:
As we are using `inlineSize/blockSize`, RDG is one step closer to handling writing mode
https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry/contentBoxSize#value

Reply:
Can we check that these values are the same as `clientWidth`/`clientHeight`?
Something like
```ts
if (clientWidth !== inlineSize || clientHeight !== blockSize) {
  throw new Error('🤔');
}
```
Just to make sure, not to be commited.

Reply:
![image](https://user-images.githubusercontent.com/1452717/162030220-74391f51-5d5d-4415-acbc-7aef70b69e8d.png)

Reply:
I guess we can give it a try, but I'm worried about decimal values leading to more shaky grids.

Reply:
If we have a reproducible example then we can give it a try